types/comparable_bytes.cc

```diff
@@ -592,11 +600,88 @@ struct to_comparable_bytes_visitor {
         out.write(bytes_view(reinterpret_cast<const signed char*>(buffer.data()), buffer_pos));
     }
 
-    // TODO: Handle other types
-
+    // Encode variable length natively byte-comparable data types.
+    // Zeros in the source are escaped as ESCAPE followed by one or more ESCAPED_0_CONT and ending with ESCAPED_0_DONE.
+    // The encoding terminates in an escaped state (either with ESCAPE or ESCAPED_0_CONT).
+    // If the source ends in a zero, we use ESCAPED_0_CONT to ensure that the encoding remains smaller than
+    // the same source with an additional zero at the end.
+    //
+    // E.g. "A\0\0B" translates to 4100FEFF4200
+    //      "A\0B\0"               4100FF4200FE
+    //      "A\0"                  4100FE
+    //      "AB"                   414200
+    //
+    // In a single-byte source, the bytes could be passed unchanged, but this would prevent combining components.
+    // This translation preserves order, and since the encoding for 0 is higher than the separator,
+    // it also ensures that shorter components are treated as smaller.
+    //
+    // The encoding is not prefix-free since, for example, the encoding of "A" (4100) is a prefix of the encoding of "A\0" (4100FE).
+    // However, the byte following the prefix is guaranteed to be FE or FF, making the encoding weakly prefix-free.
+    // Additionally, any such prefix sequence will compare as smaller than the value it prefixes,
+    // because any permitted separator byte will be smaller than the byte following the prefix.
     void operator()(const abstract_type& type) {
```

Comment:
It's not good that we have a catch-all match here. Adding a new type will not cause a compile error.

Better to list the supported types one-by-one and forward them all to a function that handles them all in the same way.